tcms/testruns/tests/test_admin.py

```diff
@@ -22,3 +29,20 @@ def test_regular_user_can_not_change_testrun(self):
         response = self.client.get(reverse('admin:testruns_testrun_change',
                                            args=[self.test_run.pk]))
         self.assertRedirects(response, reverse('testruns-get', args=[self.test_run.pk]))
+
+    def test_regular_user_can_not_delete_testrun(self):
+        response = self.client.get(reverse('admin:testruns_testrun_delete',
+                                           args=[self.test_run.pk]), follow=True)
+        self.assertContains(response, _('Permission denied: TestRun does not belong to you'))
+
+    def test_superuser_can_delete_testrun(self):
+        self.client.login(  # nosec:B106:hardcoded_password_funcarg
+            username=self.superuser.username,
+            password='password')
+        response = self.client.get(reverse('admin:testruns_testrun_delete',
+                                           args=[self.test_run.pk]))
+        self.assertContains(response, _("Yes, I'm sure"))
+        response = self.client.post(reverse('admin:testruns_testrun_delete',
+                                            args=[self.test_run.pk]), {'post': 'yes'}, follow=True)
+        self.assertRedirects(response, reverse('admin:testruns_testrun_changelist'))
+        TestRun.objects.filter(pk=self.test_run.pk).exists()
```

Comment:
Ahhh, what does this do here? 

You are asking the database for something and completely ignoring to validate the result. Regardless of the answer the test will always report PASS.